packages/components/input-number/src/input-number.vue

```diff
@@ -233,7 +233,7 @@ const setCurrentValue = (
 ) => {
   const oldVal = data.currentValue
   const newVal = verifyValue(value)
-  if (!emitChange) {
+  if (!emitChange && oldVal == newVal) {
     emit(UPDATE_MODEL_EVENT, newVal!)
     return
   }
```

Comment:
The specific reason should be that returning `''` to null in the `verifyValue` method causes oldVal and newVal to be consistent. You can consider modifying this method or adding `value !== ''` to the judgment here.

Reply:
if (oldVal === newVal && value !== '') return

Reply:
You are right. I add `value !== ''` to resolve it. 
Sorry, I pushed some unnecessary code in this PR. So I closed this pr and created [new PR](https://github.com/element-plus/element-plus/pull/14990).